Client/Extensions/Rewards/BraveLedgerExtensions.swift

```diff
@@ -144,82 +144,84 @@ extension BraveLedger {
         return
       }
       let paymentId = self.paymentId ?? ""
-      client.generateEnrollment(paymentId: paymentId, token: token) { registration, error in
-        if let error = error {
-          log.error("Failed to enroll in DeviceCheck: \(error)")
-          completion()
-          return
-        }
-        guard let registration = registration else { return }
-        client.registerDevice(enrollment: registration) { error in
-          if let error = error {
-            log.error("Failed to register device with mobile attestation server: \(error)")
-            completion()
-            return
-          }
+      Task { @MainActor in
+        do {
+          let registration = try client.generateEnrollment(paymentId: paymentId, token: token)
+          try await client.registerDevice(enrollment: registration)
+        } catch {
+          log.error("Failed to register device with mobile attestation server: \(error)")
         }
+        completion()
       }
     }
   }
 
-  public func claimPromotion(_ promotion: Ledger.Promotion, completion: @escaping (_ success: Bool) -> Void) {
+  public func claimPromotion(_ promotion: Ledger.Promotion) async -> Bool {
     guard let paymentId = self.paymentId else {
-      completion(false)
-      return
+      return false
     }
     let deviceCheck = DeviceCheckClient(environment: BraveLedger.environment)
-    let group = DispatchGroup()
     if !DeviceCheckClient.isDeviceEnrolled() {
-      group.enter()
-      setupDeviceCheckEnrollment(deviceCheck) {
-        if !DeviceCheckClient.isDeviceEnrolled() {
-          DispatchQueue.main.async {
-            completion(false)
-          }
-          return
+      let didEnrol: Bool = await withCheckedContinuation { c in
```

Comment:
typo, `didEnroll`

Reply:
It is one of those weird spellings in US vs the rest of the world 😆. For instance if I type "Enroll", macOS attempts to autocorrect it to "Enrol"

Reply:
Changed them to `enroll` spelling as thats what we've been using across the rest of the `DeviceCheckClient`